rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -489,6 +588,262 @@ impl MergeInsertJob {
         }
     }
 
+    async fn update_fragments(
+        dataset: Arc<Dataset>,
+        source: SendableRecordBatchStream,
+    ) -> Result<Vec<Fragment>> {
+        // Expected source schema: _rowaddr, updated_cols*
+        use datafusion::logical_expr::{col, lit};
+        let session_ctx = get_session_context(LanceExecutionOptions {
+            use_spilling: true,
+            ..Default::default()
+        });
+        let mut group_stream = session_ctx
+            .read_one_shot(source)?
+            .sort(vec![col(ROW_ADDR).sort(true, true)])?
```

Comment:
Hmm, clever to sort.  I wonder if we want to do that in the original merge_join?  Although I guess it changes the order of newly added rows.

Ah, I see, you are doing this for correctness and not performance.  I was thinking it might speed up the indexed take.